packages/edgeidentity/ios/src/RCTAEPEdgeIdentity.m

```diff
@@ -27,7 +29,12 @@ - (dispatch_queue_t)methodQueue
 
 RCT_EXPORT_METHOD(getExperienceCloudId:(RCTPromiseResolveBlock) resolve rejecter:(RCTPromiseRejectBlock)reject) {
     [AEPMobileEdgeIdentity getExperienceCloudId:^(NSString * _Nullable experienceCloudId, NSError * _Nullable error) {
-        resolve(experienceCloudId);
+        if (error) {
+            reject(EXTENSION_NAME, nil, error);
+            return;
+        } else {
+            resolve(experienceCloudId);
+        }
```

Comment:
would be good to return an error message as well and handle at least the timeout case, here is an example of that: 
```
if (error && error.code != AEPErrorNone) { 
	if (error.code == AEPErrorCallbackTimeout)
        reject(EXTENSION_NAME, "getExperienceCloudId call timed out", error);
            return;
    } else {
    	reject(EXTENSION_NAME, "getExperienceCloudId call returned an unexpected error", error);
    }
    return;
}

resolve(experienceCloudId);
```
You can also extract this in a helper method where you provide the API name as parameter and reuse it for the other APIs.
@yangyansong-adbe would be good to include an error description in AEPCore too that can be used for printing.

Reply:
iOS latest sample error:
WARN  Possible Unhandled Promise Rejection (id: 0):
Error: getExperienceCloudId call time out